### PR TITLE
Constrain dask and distributed versions to 2021.3.1

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

File: python/dask_cudf/setup.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
 
 install_requires = [
     "cudf",
-    "dask>=2021.3.1",
-    "distributed>=2.22.0",
+    "dask==2021.3.1",
+    "distributed==2021.3.1",
```
Comment (kkraus14) [inline python/dask_cudf/setup.py:14]: I don't think we want to change to an exact pin here, instead can we do `distributed>=2.22.0,<=2021.3.1`?

File: conda/recipes/dask-cudf/meta.yaml [hunk 2]
```diff
@@ -23,12 +23,12 @@ requirements:
   host:
     - python
     - cudf {{ version }}
-    - dask>=2021.3.1
+    - dask==2021.3.1
     - distributed >=2.22.0
```
Comment (kkraus14) [inline conda/recipes/dask-cudf/meta.yaml:27]: Need to constrain this to `>=2.22.0,<=2021.3.1` across conda recipes / environments

**Comment by jakirkham** [comment]
Should these just go into the integration repo?

**Comment by kkraus14** [comment]
> Should these just go into the integration repo?

We need them here for packaging and whatnot, but the integration package likely should pin Dask similarly.

**Review by kkraus14 — APPROVED** [review]

**Review by mike-wendt — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**